### PR TITLE
feat: add option(`useCommandShortcut`) and API(`replaceSelection`, `setSelection`, `getSelectedContent`, `deleteSelection`, `getRangeInfoOfNode`)

### PR DESCRIPTION
**Pull request by js87zz**
### Please check if the PR fulfills these requirements
- [x] It's the right issue type on the title
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] It does not introduce a breaking change or has a description of the breaking change

#### added option
* `useComamndShortcut`: if this options has `false` value, keymap would not be operated.
#### added API
* `setSelection(start: EditorPos, end: EditorPos)`: Set selection range in editor.
* `replaceSelection(content: string, start?: EditorPos, end?: EditorPos)`: Replace the content with given text content in editor. If the `start`, `end` range is passed, the content included that range would be replaced, otherwise the content of the current selection would be replaced.
* `deleteSelection(start?: EditorPos, end?: EditorPos)`: Delete the content in editor. If the `start`, `end` range is passed, the content included that range would be deleted, otherwise the content of the current selection would be deleted.
* `getSelectedContent(start?: EditorPos, end?: EditorPos)`: get the text content in editor. If the `start`, `end` range is passed, the content included that range would be returned, otherwise the content of the current selection would be returned.
* `getRangeInfoOfNode(pos?: EditorPos)`: get the range of the closest node based on specific position. If the `pos` is passed, the node range in the position would be returned, otherwise the node range in the current selection would be returned.

**addional context**
The selection range position for above APIs should be distinguished by the editor type(markdown, wysiwyg) like below.
```ts
// in markdown
start - [startLineOffset, startCharactorOffset]: [number, number], 
end - [endLineOffset, endCharactorOffset]: [number, number], 
// wysiwyg
start - startCursorOffset: number, 
end - endCursorOffset: number
```
#### Breaking changes
* remove` getTextObject` API: This API will be removed and replaced with above the new APIs
* `getRange` API => `getSelection` API: `getRange` API is replaced with `getSelection` API and the returned value would be changed as the format described above.

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

File: apps/editor/src/editorCore.ts [hunk 1]
```diff
@@ -418,32 +440,87 @@ class ToastUIEditor {
...
+   * @param {string} text - text content
    */
   insertText(text: string) {
     this.getCurrentModeEditor().replaceSelection(text);
   }
 
+  /**
+   * Set selection range
+   * @param {number|Array.<number>} [start] - start position
+   * @param {number|Array.<number>} [end] - end position
```
Comment (seonim-ryu) [inline apps/editor/src/editorCore.ts:452]: 주석에서는 `start`, `end`가 옵셔널인데 대괄호 없어도 되지 않을까요??

File: apps/editor/src/markdown/mdEditor.ts [hunk 2]
```diff
@@ -299,14 +325,28 @@ export default class MdEditor extends EditorBase {
...
+  getRangeInfoOfNode(pos?: MdPos) {
+    const { doc, selection } = this.view.state;
+    const mdPos = pos || getEditorToMdPos(doc, selection.from)[0];
+    let mdNode: MdNode = this.toastMark.findNodeAtPosition(mdPos);
+
+    if (mdNode.type === 'text' && mdNode.parent!.type !== 'paragraph') {
+      mdNode = mdNode.parent!;
+    }
+
+    mdNode.sourcepos![1][1] += 1;
```
Comment (seonim-ryu) [inline apps/editor/src/markdown/mdEditor.ts:345]: 1을 더해주는 이유는 무엇인가요~? 주석이 있으면 좋을 것 같습니다!

File: apps/editor/types/editor.d.ts [hunk 3]
```diff
@@ -208,8 +219,92 @@ export class EditorCore {
...
+
+  specs: SpecManager;
+
+  placeholder: { text: string };
+
+  createSpecs(): SpecManager;
+
+  createContext(): Context;
+
+  createState(state?: StateOptions): void;
```
Comment (seonim-ryu) [inline apps/editor/types/editor.d.ts:263]: 반환 값이 `EditorState`가 되어야하지 않나요~??

**Review by seonim-ryu — APPROVED** [review]
[3/2] 리뷰 완료합니다. 나중에 가이드에 마크다운의 셀렉션 `start` 값이 1부터 시작하는걸 잘 명시해줘야겠네 요~ 고생하셨습니다!

File: apps/editor/src/wysiwyg/wwEditor.ts [hunk 4]
```diff
@@ -226,10 +246,44 @@ export default class WysiwygEditor extends EditorBase {
...
-    const nodes = createNodesWithWidget(content, schema);
+    const nodes = createNodesWithWidget(text, schema);
+
+    this.view.dispatch(tr.replaceWith(start, end, nodes));
+  }
 
-    this.view.dispatch(tr.replaceWith(from, to, nodes));
+  getRangeInfoOfNode(pos?: number) {
+    const { doc, selection } = this.view.state;
+    const $pos = pos ? doc.resolve(pos) : selection.$from;
```
[the remark this answers is not in the document]
Comment (js87zz) [reply apps/editor/src/wysiwyg/wwEditor.ts:258]: 요거는 안그래도 한번 이야기 하고 싶었는대요 resolvedPos는 그냥 `$`프리픽스를 붙이는게 어떨까요? 그런식으로 많이 사용하더라구요.

**Pull request merged**